### PR TITLE
Remove deprecated functions in noise module

### PR DESCRIPTION
**Pull request by itoko**
### Summary
Remove deprecated functions in noise module, e.g. `NoiseModel.add_nonlocal_quantum_error`, support of json-like input for `QuantumError` and `standard_gates` argument in many `standard_errors`.
Note that `NoiseModel.from_dict` is NOT removed exceptionally until alternative methods for serialize/deserialize `NoiseModel` objects are provided.

### Details and comments
- Completely remove `noise.errors.errorutils` and `utils.noise_remapper`
- Remove deprecated methods in `NoiseModel` and `QuantumError`
- Remove deprecated argument `standard_gates` in `noise.errors.standard_errors`
- Update tests not to use deprecated functions
- Deprecate `warnings` argument in `NoiseModel` (updated from PendingDeprecationWarning)

The following PRs must be merged before this PR. 
- [x] https://github.com/Qiskit/qiskit-tutorials/pull/1397
- [x] https://github.com/Qiskit/qiskit-tutorials/pull/1400 (Update two more notebooks)
- [x] https://github.com/Qiskit/qiskit-terra/pull/9542

### CODE REVIEW

**Comment by hhorii** [comment]
@itoko Can you add a release note for removing and adding deprecation?

**Comment by itoko** [comment]
https://github.com/Qiskit/qiskit-tutorials/pull/1397 should resolve the CI failure.

**Comment by itoko** [comment]
One more relevant fix: https://github.com/Qiskit/qiskit-terra/pull/9542
Maybe we should not rush to merge this commit.

**Review by mtreinish — COMMENTED** [review]
Overall this LGTM, it's a pretty straightforward removal of stuff we deprecated some time ago. Just a few small inline comments, the only reason I didn't approve is I think we should include some more details in the release note about some of these removals.

File: releasenotes/notes/remove-deprecated-noise-functions-52128d161d3327e9.yaml [hunk 1]
```diff
@@ -0,0 +1,29 @@
...
+  - |
+    The constructor of :class:`QuantumError` has now dropped deprecated arguments,
+    ``number_of_qubits``, ``standard_gates`` and ``atol``, and the support of deprecated
+    json-like objects for ``noise_ops`` argument.
+  - |
+    The deprecated :mod:`utils.noise_remapper` and :mod:`noise.errors.errorutils` modules
+    have been entirely removed.
+  - |
+    All deprecated functions and classes
+    in :mod:`utils.noise_transformation` module has been removed.
```
[the remark this answers is not in the document]
Comment (itoko) [reply releasenotes/notes/remove-deprecated-noise-functions-52128d161d3327e9.yaml]: Addressed at [a8eb0e6](https://github.com/Qiskit/qiskit-aer/pull/1624/commits/a8eb0e6a45b262e48a90433b85002e3901f8766f)

File: qiskit_aer/noise/noise_model.py [hunk 2]
```diff
@@ -1062,6 +902,43 @@ def from_dict(noise_dict):
         warn('from_dict has been deprecated as of qiskit-aer 0.10.0'
              ' and will be removed no earlier than 3 months from that release date.',
              DeprecationWarning, stacklevel=2)
+
+        # pylint: disable=import-outside-toplevel
+        from qiskit.circuit import QuantumCircuit
+        from qiskit.extensions import UnitaryGate
```
[the remark this answers is not in the document]
Comment (itoko) [reply qiskit_aer/noise/noise_model.py]: Fixed at [4203f52](https://github.com/Qiskit/qiskit-aer/pull/1624/commits/4203f52dde9584fc268f067398d48c58cddaa7d9)

**Review by hhorii — APPROVED** [review]

**Pull request merged**